### PR TITLE
First tries to recenter user model to origin  

### PR DESCRIPTION
**Pull request by natacha-beck**
The idea here was to recenter the model to the origin if the original data is shifted. In this way the user data should appear at the center when it is loaded. 
I tried to see all the edges case that can appear in the demo, but maybe I miss some. 

Can you take a second look and let me know if you think it is ok.

### CODE REVIEW

File: examples/surface-viewer-demo.html [hunk 1]
```diff
@@ -192,6 +192,11 @@ <h3 id="annotation-header">Your Annotations</h3>
                 <input id="axes-controls" type="checkbox">Display axes<br />
               </p>
               <p>
+                <div>
+                  <input type="checkbox"  id="model_centric"><label for="model_centric">Origin correspond to model center</label>
```
Comment (prioux) [inline examples/surface-viewer-demo.html]: Missing two "s":

> Origin corresponds to model's center

File: src/brainbrowser/surface-viewer/modules/rendering.js [hunk 2]
```diff
@@ -209,12 +210,12 @@ BrainBrowser.SurfaceViewer.modules.rendering = function(viewer) {
...
-            centroid.x,
-            centroid.y,
-            centroid.z
+            centroid.x + offset.x,
+            centroid.y + offset.y,
+            centroid.z + offset.z
           );
         } else {
-          shape.position.set(0, 0, 0);
+          shape.position.set(0 + offset.x, 0 + offset.y, 0 + offset.z);
```
Comment (prioux) [inline src/brainbrowser/surface-viewer/modules/rendering.js]: Why add zeros ?

File: src/brainbrowser/surface-viewer/modules/rendering.js [hunk 3]
```diff
@@ -269,10 +270,16 @@ BrainBrowser.SurfaceViewer.modules.rendering = function(viewer) {
     var geometry = new THREE.SphereGeometry(radius);
     var material = new THREE.MeshBasicMaterial({color: color});
 
-    var sphere = new THREE.Mesh(geometry, material);
+    var sphere   = new THREE.Mesh(geometry, material);
     sphere.position.set(x, y, z);
 
     if (viewer.model) {
+      var offset = viewer.model.userData.model_center_offset;
+      if (offset !== undefined) {
```
Comment (prioux) [inline src/brainbrowser/surface-viewer/modules/rendering.js]: so offsetting always happen, even when the checkbox is off ?

File: src/brainbrowser/surface-viewer/modules/rendering.js [hunk 4]
```diff
@@ -623,6 +630,98 @@ BrainBrowser.SurfaceViewer.modules.rendering = function(viewer) {
...
+    offset_centroid.copy(model.userData.model_center_offset);
+    if (model_centric === false) { offset_centroid.negate();}
+
+    model.children.forEach(function(children) {
+      // Return if children is not given by the user
+      if (Object.keys(children.userData).length === 0 && children.userData.constructor === Object) {return;}
+      children.translateX(offset_centroid.x);
+      children.translateY(offset_centroid.y);
+      children.translateZ(offset_centroid.z);
+      model.userData.model_centric = model_centric;
```
Comment (prioux) [inline src/brainbrowser/surface-viewer/modules/rendering.js]: why is this assignment in the loop?

File: src/brainbrowser/surface-viewer/modules/rendering.js [hunk 5]
```diff
@@ -623,6 +631,99 @@ BrainBrowser.SurfaceViewer.modules.rendering = function(viewer) {
...
+  * @description
+  * Use to recenter data when userData input is shifted in space.
+  *
+  *
+  * ```js
+  * viewer.modelCentric(true);
+  * ```
+  */
+  viewer.modelCentric = function(model_centric) {
+    if (model_centric === undefined) {model_centric = false;}
```
Comment (rdvincent) [inline src/brainbrowser/surface-viewer/modules/rendering.js]: Sorry if this is overly pedantic, but I really prefer to keep statements on separate lines:
```
if (model_centric === undefined) {
   model_centric = false;
}
```

Comment (natacha-beck) [inline src/brainbrowser/surface-viewer/modules/rendering.js]: Ok I will fix it, and I'm kind of agree with you. 
In fact I like used `return if something` but it is not supported in JS.

File: src/brainbrowser/surface-viewer/modules/rendering.js [hunk 6]
```diff
@@ -623,6 +631,107 @@ BrainBrowser.SurfaceViewer.modules.rendering = function(viewer) {
...
+    viewer.findUserDataCentroid(model);
+
+    if (model_centric === model.userData.model_centric) {
+      return;
+    }
+
+    // Caculate the offset
+    var offset_centroid = new THREE.Vector3();
+    offset_centroid.copy(model.userData.model_center_offset);
+    if (model_centric === false) { offset_centroid.negate();}
```
Comment (rdvincent) [inline src/brainbrowser/surface-viewer/modules/rendering.js]: Missed one here.

File: src/brainbrowser/surface-viewer/modules/rendering.js [hunk 7]
```diff
@@ -623,6 +631,107 @@ BrainBrowser.SurfaceViewer.modules.rendering = function(viewer) {
...
+
+    model.children.forEach(function(children) {
+      // Return if children is not given by the user
+      if (Object.keys(children.userData).length === 0 && children.userData.constructor === Object) {return;}
```
Comment (rdvincent) [inline src/brainbrowser/surface-viewer/modules/rendering.js]: ...and here.

**Comment by rdvincent** [comment]
Looks good. Shall I merge?

**Comment by natacha-beck** [comment]
If it seems ok for you yes Pierre had review too, so everything seems ok, I
will do a release next week.

**Pull request merged**